EU4toV2/Source/V2World/Country/Country.h

```diff
@@ -0,0 +1,166 @@
+		void setColonyOverlord(std::shared_ptr<Country> colony) { colonyOverlord = colony; }
+		int getNumFactories() const { return details.numFactories; }
+		Relation& getRelation(const std::string& withWhom);
+		void addRelation(Relation& newRelation);
+		void absorbVassal(std::shared_ptr<Country> vassal);
+		const std::map<int, std::shared_ptr<Province>>& getProvinces() const { return provinces; }
+		void addPrestige(double additionalPrestige) { details.prestige += additionalPrestige; }
+		double getPrestige() const { return details.prestige; }
+		std::tuple<double, double, double> getNationalValueScores();
+		void setNationalValue(const std::string& NV) { details.nationalValue = NV; }
+		std::vector<std::shared_ptr<State>> getStates() const { return states; }
+		bool isCivilized() const { return details.civilized; }
+		void addProvince(std::shared_ptr<Province> _province);
+		int getCapital() const { return details.capital; }
+		void addState(std::shared_ptr<State> newState, const mappers::PortProvinces& portProvincesMapper);
+		void convertLeaders(mappers::LeaderTraitMapper& leaderTraitMapper);
+		void convertUncivReforms(int techGroupAlgorithm, double topTech, int topInstitutions, const mappers::TechGroupsMapper& techGroupsMapper);
+		void addRailroadtoCapitalState();
+		void addTech(const std::string& newTech) { techs.insert(newTech); }
+		void convertLandlessReforms(std::shared_ptr<Country> capOwner);
+		const UncivReforms& getUncivReforms() const { return uncivReforms; }
+		void setArmyTech(double normalizedScore);
+		void setNavyTech(double normalizedScore);
+		void setCommerceTech(double normalizedScore);
+		void setIndustryTech(double normalizedScore);
+		void setCultureTech(double normalizedScore);
+		bool addFactory(std::shared_ptr<Factory> factory);
+		std::string getCommonCountryFile() const { return commonCountryFile; }
+		std::string getLocalName() const { return localisation.getLocalName(); };
+		std::string getReligion() const { return details.religion; }
+		std::string getColonialRegion();
+		const Localisation& getLocalisation() const { return localisation; }
+		bool isDynamicCountry() const { return dynamicCountry; }
+		std::string getFilename() const { return details.filename; }
+		static bool factoryCandidateSortPredicate(const std::pair<double, std::shared_ptr<State>>& lhs, const std::pair<double, std::shared_ptr<State>>& rhs);
+
+		friend std::ostream& operator<<(std::ostream& output, const Country& country);
+		void outputCommons(std::ostream& output);
+		void outputOOB(std::ostream& output);
+		
+	private:
+		bool dynamicCountry = false;	// true if this country is a Vic2 dynamic country
+		bool newCountry = false; // true if this country is being added by the converter, i.e. doesn't already exist in Vic2
+		mappers::CountryDetails details;
+		std::string tag;
+		std::string commonCountryFile;
+		Localisation localisation;
+		std::shared_ptr<EU4::Country> srcCountry;
+		UncivReforms uncivReforms;
```

Comment:
An alternate to making it store whether its initialized (and both getting default-constructed and then recreated) would be to have it as a shared_ptr or optional. Optional might make more semantic sense.

Also applies other similar items.

Reply:
yes, this is much better. fixed.